src/shims/unix/foreign_items.rs

```diff
@@ -315,6 +315,41 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                     }
                 }
             }
+            "aligned_alloc" => {
+                let [align, size] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let align = this.read_target_usize(align)?;
+                let size = this.read_target_usize(size)?;
+
+                // The size must be a power of the alignment.
+                // The specs says nothing about error handling other than null
+                // pointer but aligned_alloc Linux/macOs implementations set EINVAL. Note that
+                // on FreeBSD setting a size not being a power of alignment is UB but we do not
+                // want it.
```

Comment:
Ah no actually it can't be UB

> Passing a size which is not an integral multiple of alignment or an alignment which is not valid or not supported by the implementation causes the function to fail and return a null pointer

Is it truly UB in FreeBSD? That's a critical bug in their implementation then, may be worth reporting.

But, we shouldn't set_last_error, as the spec says nothing like that.